tests/src/Content/Text/BBCodeTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Friendica\Test\src\Content\Text;
+
+use Friendica\Content\Text\BBCode;
+use Friendica\Test\MockedTest;
+use Friendica\Test\Util\AppMockTrait;
+use Friendica\Test\Util\L10nMockTrait;
+use Friendica\Test\Util\VFSTrait;
+
+/**
+ * @runTestsInSeparateProcesses
+ * @preserveGlobalState disabled
+ */
+class BBCodeTest extends MockedTest
+{
+	use VFSTrait;
+	use AppMockTrait;
+	use L10nMockTrait;
+
+	protected function setUp()
+	{
+		parent::setUp();
+
+		$this->setUpVfsDir();
+		$this->mockApp($this->root);
+
+		$this->app->videowidth = 425;
+		$this->app->videoheight = 350;
+
+		$this->configMock->shouldReceive('get')
+			->with('system', 'remove_multiplicated_lines')
+			->andReturn(false);
+		$this->configMock->shouldReceive('get')
+			->with('system', 'no_oembed')
+			->andReturn(false);
+		$this->configMock->shouldReceive('get')
+			->with('system', 'allowed_link_protocols')
+			->andReturn(null);
+		$this->configMock->shouldReceive('get')
+			->with('system', 'itemcache_duration')
+			->andReturn(-1);
+
+		$this->mockL10nT();
+	}
+
+	public function dataLinks()
+	{
+		return [
+			/** @see https://github.com/friendica/friendica/issues/2487 */
+			'bug-2487-1' => [
+				'data' => 'https://de.wikipedia.org/wiki/Juha_Sipilä',
+				'link' => true,
+			],
+			'bug-2487-2' => [
+				'data' => 'https://de.wikipedia.org/wiki/Dnepr_(Motorradmarke)',
+				'link' => true,
+			],
+			'bug-2487-3' => [
+				'data' => 'https://friendica.wäckerlin.ch/friendica',
+				'link' => false,
+			],
+			'bug-2487-4' => [
+				'data' => 'https://mastodon.social/@morevnaproject',
+				'link' => false,
```

Comment:
This should be true, this is a valid link.

Reply:
when I set it to true:
```console
Failed asserting that two strings are equal.
Expected :'<a href="https://mastodon.social/@morevnaproject" target="_blank">https://mastodon.social/@morevnaproject</a>'
Actual   :'https://mastodon.social/@morevnaproject'
```